src/tests/system/tests/test_passkey.py

```diff
@@ -241,3 +241,188 @@ def test_passkey__su_srv_not_resolvable(
+    client.sssd.bring_offline()
+
+    assert client.auth.su.passkey(
+        username="user1",
+        pin=123456,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script.{suffix}",
+    )
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.builtwith(client="passkey", provider="passkey")
+def test_passkey__user_fetch_from_cache(
+    client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str
+):
+    """
+    :title: Fetch a user from cache for LDAP, IPA, AD and Samba server
+    :setup:
+        1. Add a user in LDAP, IPA, AD and Samba with passkey_mapping.
+        2. Setup SSSD client with FIDO and umockdev, start SSSD service.
+    :steps:
+        1. Check a user lookup.
+        2. Check a user from cache using ldbsearch command.
+    :expectedresults:
+        1. A user looked up successfully.
+        2. Successfully get the user from ldbsearch command.
+    :customerscenario: False
+    """
+
+    suffix = type(provider).__name__.lower()
+
+    if suffix == "ldap":
+        client.sssd.domain["local_auth_policy"] = "only"
+
+    with open(f"{testdatadir}/passkey-mapping.{suffix}") as f:
+        provider.user("user1").add().passkey_add(f.read().strip())
+
+    client.sssd.start()
+
+    result = client.tools.id("user1")
+    output = client.ldb.search(path="/var/lib/sss/db/cache_test.ldb", basedn="cn=test,cn=sysdb")
+    assert result is not None
+    assert output["name=user1@test,cn=users,cn=test,cn=sysdb"] is not None
+    assert "user1" in output.stdout_lines[1]
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.builtwith(client="passkey", provider="passkey")
+def test_passkey__su_multi_keys_for_same_user(
+    client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str
+):
+    """
+    :title: Check su authentication of user when multiple keys added for same user with
+            LDAP, IPA, AD and Samba server.
+    :setup:
+        1. Add a user with multiple mappings of passkey in LDAP, IPA, AD and Samba with passkey_mapping.
+        2. Setup SSSD client with FIDO and umockdev, start SSSD service.
+    :steps:
+        1. Check su authentication of the user.
+    :expectedresults:
+        1. User su authenticates successfully.
+    :customerscenario: False
+    """
+    suffix = type(provider).__name__.lower()
+    user_add = provider.user("user1").add()
+
+    if suffix == "ldap":
+        client.sssd.domain["local_auth_policy"] = "only"
+        for n in range(1, 5):
+            if bool(user_add.get(attrs=["passkey"])):
+                with open(f"{testdatadir}/passkey-mapping.{suffix}{n}") as f:
+                    attrs = {"passkey": f.read().strip()}
+                    user_add._modify(add=attrs)
```

Comment:
The leading underscore in `_modify` means that this is an internal use function, meaning that it should only be use inside the `LDAPObject`. Thus, no external use.

In addition, you must use the `passkey_add` API to add the keys to the LDAP attribute. If you run into any problems using it, then you must overcome this problem within `passkey_add` so that any other user of this interface does not have to create another ad-hoc solution.

Reply:
okay, but in ldap while adding keys and `objectClass`, first time we need to add `objectClass` with keys but second time we cant do that with same method and second issue was we need to maintain the sequence the key mapping sequence which we had while while creating the recording files, if we missed that authentication failed, else it wont need that way.

Reply:
Since https://github.com/SSSD/sssd-test-framework/pull/66 is merged, you can update this part of the code.

Reply:
updated.